op-node/rollup/derive/frame.go

```diff
@@ -130,6 +133,29 @@ func ParseFrames(data []byte) ([]Frame, error) {
 	if len(data) == 0 {
 		return nil, errors.New("data array must not be empty")
 	}
+	// If the calldata starts with the function signature of the submit method
+	// then it is a v2 frame.
+	// TODO: don't parse the ABI every time
+	abi, err := bindings.LeaderElectionBatchInboxMetaData.GetAbi()
+	if err != nil || abi == nil {
+		return nil, fmt.Errorf("could not get abi: %w", err)
+	}
+	submit, exists := abi.Methods["submit"]
+	if !exists {
+		return nil, errors.New("could not find submit method in bindings")
+	}
+	// Check if the signature matches the submit function
+	// TODO: v2 code currently untested
+	isV2 := len(data) >= 4 && bytes.Equal(data[:4], submit.ID)
+	var metas []bindings.LeaderElectionBatchInboxMeta
+	if isV2 {
+		metas, data, err = ParseFramesV2(data, submit)
+		if err != nil {
+			return nil, fmt.Errorf("parsing v2 frames: %w", err)
+		}
+	}
+
+	// Otherwise it is a v0 frame.
```

Comment:
This v0 is not related to the V1/V2 on the Batch.

We're still DerivationVersion0. Which seems to be consistent with the code below, but this comment is misleading.

Reply:
Thanks. Updated the comments.